proof_of_play.go

```diff
@@ -158,3 +168,58 @@ func (p *proofOfPlay) expire(ad Ad) error {
 		return false, nil
 	})
 }
+
+func isLeaseExpired(ad Ad) bool {
+	exp, ok := ad["lease_expiry"].(float64)
+	if !ok {
+		return true
```

Comment:
From the KB it's always present in the response, but really it's up to the the caller to pass in this field. I think it should be required since otherwise we'll be retrying this infinitely.